examples/ui/main.rs

```diff
@@ -176,48 +176,28 @@ impl State for Example {
             ))
             .build();
 
-        // Manual button creation example
-        let button_image = world
-            .create_entity()
-            .with(UiTransform::new(
-                "btn_image".to_string(),
-                0.0,
-                32.0,
-                -1.0,
-                128.0,
-                64.0,
-                9,
-            ))
-            .with(UiImage {
-                texture: green.clone(),
-            })
-            .with(Anchored::new(Anchor::TopMiddle))
-            .with(Parent {
-                entity: background.clone(),
-            })
-            .with(MouseReactive)
-            .build();
-        world
-            .create_entity()
-            .with(UiTransform::new(
-                "btn_text".to_string(),
-                0.,
-                0.,
-                -1.,
-                0.,
-                0.,
-                10,
-            ))
-            .with(UiText::new(
+        let _button = UiButtonBuilder::new("btn", "Button!", world)
+            .with_uitext(UiText::new(
                 font.clone(),
                 "Button!".to_string(),
                 [0.2, 0.2, 1.0, 1.0],
                 20.,
             ))
-            .with(Anchored::new(Anchor::Middle))
-            .with(Stretched::new(Stretch::XY, 0.0, 0.0))
-            .with(Parent {
-                entity: button_image,
+            .with_transform(UiTransform::new(
+                    "btn_transform".to_string(),
+                    0.0,
+                    32.0,
+                    -1.0,
+                    128.0,
+                    64.0,
+                    9,
+            ))
+            .with_image(UiImage {
+                texture: green.clone(),
+            })
+            .with_anchored(Anchored::new(Anchor::TopMiddle))
+            .with_parent(Parent {
+                entity: background.clone(),
             })
             .build();
 
```

Comment:
Could you add another button showing a less heavy syntax?
e.g: UiButtonBuilder::new("btn2", "Simple Button").with_font(font.clone()).with_position(50.0,50.0).build(&mut world);